src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -120,7 +129,37 @@ export class StacheNavService {
       .join(' ');
   }
 
-  private clone(thing: any): any {
-    return JSON.parse(JSON.stringify(thing));
+  private sortRoutes(routes: StacheNavLink[]): StacheNavLink[] {
+
+    const sortByName = (a: any, b: any): number  => {
+      if (a.name.toLowerCase() < b.name.toLowerCase()) {
+        return -1;
+      } else if (a.name.toLowerCase() > b.name.toLowerCase()) {
```

Comment:
That's correct, however to get tests/lint to go over all the "branch" options it gets frustrated with if statements without else/else ifs.    Also, the `sort` method whenever you look them up, all seem to be written in this format.   #reasonsTestsRuinCode